### PR TITLE
upgrade service daos to sqlalchemy 2.0

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

Upgrade query syntax to sqlalchemy 2.0 in certain selected files.

## Security Considerations

N/A

### CODE REVIEW

**Review by xlorepdarkhelm — APPROVED** [review]

File: app/dao/service_permissions_dao.py [hunk 1]
```diff
@@ -16,9 +21,17 @@ def dao_add_service_permission(service_id, permission):
...
+
+    stmt = delete(ServicePermission).where(
         ServicePermission.service_id == service_id,
         ServicePermission.permission == permission,
-    ).delete()
+    )
+    result = db.session.execute(stmt)
     db.session.commit()
-    return deleted
+    return result.rowcount
```
[the remark this answers is not in the document]
Comment (terrazoon) [reply app/dao/service_permissions_dao.py:37]: yes, that's correct.

File: app/dao/service_permissions_dao.py [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 from app import db
 from app.dao.dao_utils import autocommit
 from app.models import ServicePermission
 
 
 def dao_fetch_service_permissions(service_id):
-    return ServicePermission.query.filter(
-        ServicePermission.service_id == service_id
-    ).all()
+    # return ServicePermission.query.filter(
```
Comment (stvnrlly) [inline app/dao/service_permissions_dao.py:9]: Okay to remove the previous version now?

File: app/dao/service_permissions_dao.py [hunk 3]
```diff
@@ -16,9 +21,17 @@ def dao_add_service_permission(service_id, permission):
 
 
 def dao_remove_service_permission(service_id, permission):
-    deleted = ServicePermission.query.filter(
+    # deleted = ServicePermission.query.filter(
```
Comment (stvnrlly) [inline app/dao/service_permissions_dao.py:24]: Okay to remove the previous version now?

File: app/dao/service_user_dao.py [hunk 4]
```diff
@@ -7,19 +9,28 @@ def dao_get_service_user(user_id, service_id):
...
+    #     user_id=user_id, service_id=service_id
+    # ).one_or_none()
+    stmt = select(ServiceUser).filter_by(user_id=user_id, service_id=service_id)
+    return db.session.execute(stmt).scalars().one_or_none()
 
 
 def dao_get_active_service_users(service_id):
-    query = (
-        db.session.query(ServiceUser)
+    # query = (
```
Comment (stvnrlly) [inline app/dao/service_user_dao.py:20]: Okay to remove the previous version now?

**Review by stvnrlly — APPROVED** [review]

**Pull request merged**